pandas/tests/indexes/test_label_or_level_utils.py

```diff
@@ -0,0 +1,378 @@
+import pytest
+import pandas as pd
+from pandas import compat
+import pandas.util.testing as tm
+from pandas.core.dtypes.missing import array_equivalent
+
+
+class TestLabelOrLevelUtils(object):
+    """
+    Test NDFrame utility methods used by operations that allow users to
+    specify a mixture of levels and labels
+    """
+
+    # Setup
+    # =====
```

Comment:
I would prefer you use fixtures here and not the old style steup_method. (e.g. this is what prepare_df basically is already)

Reply:
move this to pandas/tests/generic (new dir I am creating now)